File: icons/gerar-icones.py
#!/usr/bin/env python3
"""
Script para gerar ícones do PWA
Requer: pip install Pillow
Execute: python icons/gerar-icones.py
"""

try:
    from PIL import Image, ImageDraw, ImageFont
    import os
except ImportError:
    print("⚠️  Pillow não está instalado.")
    print("Instale com: pip install Pillow")
    exit(1)

def criar_icone(tamanho, output_path):
    """Cria um ícone PNG com o tamanho especificado"""
    # Criar imagem com gradiente
    img = Image.new('RGB', (tamanho, tamanho), color='#667eea')
    draw = ImageDraw.Draw(img)
    
    # Desenhar fundo com gradiente diagonal
    for y in range(tamanho):
        for x in range(tamanho):
            # Gradiente diagonal
            ratio = (x + y) / (tamanho * 2)
            r1, g1, b1 = 102, 126, 234  # #667eea
            r2, g2, b2 = 118, 75, 162   # #764ba2
            r = int(r1 + (r2 - r1) * ratio)
            g = int(g1 + (g2 - g1) * ratio)
            b = int(b1 + (b2 - b1) * ratio)
            draw.point((x, y), fill=(r, g, b))
    
    # Criar máscara para bordas arredondadas
    mask = Image.new('L', (tamanho, tamanho), 0)
    mask_draw = ImageDraw.Draw(mask)
    radius = int(tamanho * 0.2)
    
    # Desenhar retângulo arredondado na máscara
    mask_draw.rounded_rectangle([(0, 0), (tamanho, tamanho)], radius=radius, fill=255)
    
    # Aplicar máscara ao fundo
    img.putalpha(mask)
    
    # Converter para RGB final
    final_img = Image.new('RGB', (tamanho, tamanho), (255, 255, 255))
    final_img.paste(img, (0, 0), img.split()[3] if img.mode == 'RGBA' else None)
    
    draw = ImageDraw.Draw(final_img)
    
    # Tentar carregar fonte, usar padrão se não encontrar
    try:
        # Tentar usar fonte do sistema
        font_size_main = int(tamanho * 0.35)
        font_size_sub = int(tamanho * 0.15)
        try:
            font_main = ImageFont.truetype("arial.ttf", font_size_main)
            font_sub = ImageFont.truetype("arial.ttf", font_size_sub)
        except:
            try:
                font_main = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", font_size_main)
                font_sub = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", font_size_sub)
            except:
                font_main = ImageFont.load_default()
                font_sub = ImageFont.load_default()
    except:
        font_main = ImageFont.load_default()
        font_sub = ImageFont.load_default()
    
    # Texto principal "E365"
    text_main = "E365"
    bbox = draw.textbbox((0, 0), text_main, font=font_main)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]
    draw.text(
        ((tamanho - text_width) / 2, tamanho / 2 - tamanho * 0.1 - text_height / 2),
        text_main,
        fill='white',
        font=font_main
    )
    
    # Texto secundário "English"
    text_sub = "English"
    bbox = draw.textbbox((0, 0), text_sub, font=font_sub)
    text_width = bbox[2] - bbox[0]
    draw.text(
        ((tamanho - text_width) / 2, tamanho / 2 + tamanho * 0.05),
        text_sub,
        fill=(255, 255, 255, 230),
        font=font_sub
    )
    
    # Salvar
    final_img.save(output_path, 'PNG')
    print(f"✓ Ícone {tamanho}x{tamanho} criado: {output_path}")

def main():
    # Obter diretório do script
    script_dir = os.path.dirname(os.path.abspath(__file__))
    
    print("Gerando ícones do PWA...\n")
    
    # Criar ícones
    criar_icone(192, os.path.join(script_dir, 'icon-192.png'))
    criar_icone(512, os.path.join(script_dir, 'icon-512.png'))
    
    print("\n✓ Todos os ícones foram criados com sucesso!")
    print(f"Localização: {script_dir}")

if __name__ == '__main__':
    main()

File: icons/gerar-icones.js
// Script Node.js para gerar ícones do PWA
// Execute: node icons/gerar-icones.js

const fs = require('fs');
const path = require('path');

// Função para criar ícone PNG usando canvas (requer canvas npm package)
// Se não tiver canvas instalado, use o gerar-icones.html no navegador

function criarIconeBase64(tamanho) {
    // Criar um canvas virtual (requer: npm install canvas)
    try {
        const { createCanvas } = require('canvas');
        const canvas = createCanvas(tamanho, tamanho);
        const ctx = canvas.getContext('2d');
        
        // Gradiente de fundo
        const gradient = ctx.createLinearGradient(0, 0, tamanho, tamanho);
        gradient.addColorStop(0, '#667eea');
        gradient.addColorStop(1, '#764ba2');
        
        // Fundo com bordas arredondadas
        ctx.fillStyle = gradient;
        ctx.beginPath();
        const radius = tamanho * 0.2;
        ctx.moveTo(radius, 0);
        ctx.lineTo(tamanho - radius, 0);
        ctx.quadraticCurveTo(tamanho, 0, tamanho, radius);
        ctx.lineTo(tamanho, tamanho - radius);
        ctx.quadraticCurveTo(tamanho, tamanho, tamanho - radius, tamanho);
        ctx.lineTo(radius, tamanho);
        ctx.quadraticCurveTo(0, tamanho, 0, tamanho - radius);
        ctx.lineTo(0, radius);
        ctx.quadraticCurveTo(0, 0, radius, 0);
        ctx.closePath();
        ctx.fill();
        
        // Texto principal
        ctx.fillStyle = 'white';
        ctx.font = `bold ${tamanho * 0.35}px Arial`;
        ctx.textAlign = 'center';
        ctx.textBaseline = 'middle';
        ctx.fillText('E365', tamanho / 2, tamanho / 2 - tamanho * 0.1);
        
        // Texto secundário
        ctx.font = `${tamanho * 0.15}px Arial`;
        ctx.fillStyle = 'rgba(255, 255, 255, 0.9)';
        ctx.fillText('English', tamanho / 2, tamanho / 2 + tamanho * 0.15);
        
        // Salvar como PNG
        const buffer = canvas.toBuffer('image/png');
        const filename = path.join(__dirname, `icon-${tamanho}.png`);
        fs.writeFileSync(filename, buffer);
        console.log(`✓ Ícone ${tamanho}x${tamanho} criado: ${filename}`);
        
        return true;
    } catch (error) {
        console.log('Canvas não disponível. Use o gerar-icones.html no navegador.');
        console.log('Ou instale: npm install canvas');
        return false;
    }
}

// Gerar ícones
console.log('Gerando ícones do PWA...\n');
const icon192 = criarIconeBase64(192);
const icon512 = criarIconeBase64(512);

if (icon192 && icon512) {
    console.log('\n✓ Todos os ícones foram criados com sucesso!');
} else {
    console.log('\n⚠ Use o arquivo gerar-icones.html no navegador para gerar os ícones.');
}
